### PR TITLE
handle dropped transactions

### PR DESCRIPTION
**Pull request by estebanmino**
1. I was able to repro a case where speeding / cancelling txs got a tx not broadcasted and stuck on the tx list for ever
image

it's fixed by checking the last confirmed tx nonce, if is higher than the stuck tx nonce it will  change that status to unknown and it won't be saved on state

2. When you close the app with a pending transaction, that transaction is stuck in "Pending" until a new transaction is sent so I added a change to watch transactions on wallet initialized, if there's no pending tx it won't do anything
image
https://github.com/rainbow-me/rainbow/compare/@esteban/handle-dropped-transactions?expand=1#diff-12d8f761ac899556e7c0f7538b03816a6e871c1105b73debf28c1ba76f63ac97R135

fixes https://github.com/rainbow-me/rainbow/issues/2396

### CODE REVIEW

File: src/redux/data.js [hunk 1]
```diff
@@ -830,6 +834,8 @@ export const dataWatchPendingTransactions = (
               type: tx?.type,
             });
             updatedPending.status = newStatus;
+          } else if (currentNonce > tx.nonce) {
+            updatedPending.status = TransactionStatusTypes.unknown;
```
Comment (brunobar79) [inline src/redux/data.js:838]: should we throw a custom error to sentry to know how often this happens?

**Review by brunobar79 — APPROVED** [review]
💯

**Review by jinchung — CHANGES_REQUESTED** [review]
Marking this as "request changes" to hold off on QA / merge until some questions are answered (separate thread on Slack)

File: src/redux/data.js [hunk 2]
```diff
@@ -904,7 +914,20 @@ const updatePurchases = updatedTransactions => dispatch => {
   dispatch(addCashUpdatePurchases(confirmedPurchases));
 };
 
-const watchPendingTransactions = (
+export const watchPendingTransactionsOnInitialize = (
```
Comment (jinchung) [inline src/redux/data.js:917]: confirming if this is not going to be "watching" and just checking once on init, then can we rename to `watchPendingTransactionsOnInitialize`

File: src/redux/data.js [hunk 3]
```diff
@@ -794,23 +794,27 @@ export const dataWatchPendingTransactions = (provider = null) => async (
...
+        // if the nonce of last confirmed tx is higher than this pending tx then it got dropped
+        const nonceAlreadyIncluded = currentNonce > tx.nonce;
+        if (
+          (txObj && txObj.blockNumber && txObj.blockHash) ||
+          nonceAlreadyIncluded
+        ) {
           // When speeding up a non "normal tx" we need to resubscribe
           // because zerion "append" event isn't reliable
           logger.log('TX CONFIRMED!', txObj);
           appEvents.emit('transactionConfirmed', txObj);
```
Comment (jinchung) [inline src/redux/data.js:813]: do we want to not emit this for when `nonceAlreadyIncluded`

File: src/redux/data.js (hunk 2 above)
Comment (estebanmino) [reply src/redux/data.js:917]: yes, only once

**Review by jinchung — APPROVED** [review]
🌮

**Review by skylarbarrera — APPROVED** [review]
👌

**Pull request merged**